### PR TITLE
Closes #221 Removes suggested packages and update CI actions

### LINKED ISSUE #221: Q: Can all or any of the Suggested packages be removed?
Thinking about:

- {devtools}
- {lintr}
- {spelling}
- {usethis}
  - only 1 mention on documentation

### PR DESCRIPTION
**Pull request by averissimo**
- Closes #221 

### Changes Description

- Removed dependencies from `DESCRIPTION`
- Removed `SASxport` installation in CI
- Adds items to wordlist (`spelling` action doesn't pass without :smile:)
- Using same CI/CD infrastructure as Admiral
  - Except for `R CMD check` to test multiple platforms and R versions

### Task List

- [x] The spirit of xportr is met in your Pull Request
- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Summary of changes filled out in the above Changes Description. Can be removed or left blank if changes are minor/self-explanatory.
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Use `styler` package and functions to style files accordingly.
- [x] Updated relevant unit tests or have written new unit tests. See our [Wiki](https://github.com/atorus-research/xportr/wiki/Style-Guide-for-Unit-Tests) for conventions used in this package.
- [x] Creation/updated relevant roxygen headers and examples. See our [Wiki](https://github.com/atorus-research/xportr/wiki/Style-Guide-for-Roxygen-Headers) for conventions used in this package.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Run `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new/updated functions occur on the "Reference" page.
- [x] Update NEWS.md if the changes pertain to a user-facing function (i.e. it has an @export tag) or documentation aimed at users (rather than developers)
- [x] Make sure that the package version in the NEWS.md and DESCRIPTION file is same. Don't worry about updating the version because it will be auto-updated using the `vbump.yaml` CI.
- [x] Address any updates needed for vignettes and/or templates
- [x] Link the issue Development Panel so that it closes after successful merging.
- [x] Fix merge conflicts
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

**Comment by averissimo** [comment]
It's difficult to test this, so I've added a commit that will fail the tests.

Feel free to add and remove commits to test PR

- 74554866450ed6530b3a3a6097bd2869f5e94e74 passes CI (excluding links)
- 40d4e80235294ca3a4929c01819c380203c7934d (fails spelling / linter / styling CI)
- a649c198dbd46195cb5231bd4bfde205ece8c99f (back to passing tests)

File: .github/workflows/check-links.yml [hunk 1]
```diff
@@ -7,15 +8,22 @@ on:
   pull_request:
     branches:
       - main
+  workflow_dispatch:
```
Comment (averissimo) [inline .github/workflows/check-links.yml:11]: Allows user to trigger CI from github UI for  a given branch.

image

File: .github/workflows/check-links.yml [hunk 2]
```diff
@@ -7,15 +8,22 @@ on:
...
+
+concurrency:
+  group: links-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
```
Comment (averissimo) [inline .github/workflows/check-links.yml:15]: Doesn't allow for the same workflow/action to run multiple times in the same PR.

Doesn't apply to `R CMD check` workflow nor `vbump`

File: README.Rmd [hunk 3]
```diff
@@ -46,7 +46,7 @@ install.packages("xportr")
 ### Development version:
 
 ```{r, eval = FALSE}
-devtools::install_github("https://github.com/atorus-research/xportr.git", ref = "devel")
+install.packages("teal", repos = c("https://pharmaverse.r-universe.dev", getOption("repos")))
```
Comment (elimillera) [inline README.Rmd:49]: Should 'teal' be updated to 'xportr' here? I'm not sure if 'xportr' is hosted on the pharmaverse.r-universe

File: .github/workflows/spellcheck.yml [hunk 4]
```diff
@@ -14,50 +15,26 @@ concurrency:
...
-        run: |
-          while read -r cmd
-          do
-            eval sudo $cmd
-          done < <(Rscript -e 'writeLines(remotes::system_requirements("ubuntu", "20.04"))')
+          extra-packages: any::spelling
 
       - name: Run Spellcheck 👟
-        uses: insightsengineering/r-spellcheck-action@v3
+        uses: insightsengineering/r-spellcheck-action@v2
```
Comment (elimillera) [inline .github/workflows/spellcheck.yml:37]: Is this downgrade from v3 intentional?

File: README.Rmd (hunk 3 above)
Comment (averissimo) [reply README.Rmd:49]: Oh sorry! You're absolutely right.

It is hosted there: https://pharmaverse.r-universe.dev/xportr

File: .github/workflows/spellcheck.yml (hunk 4 above)
Comment (averissimo) [reply .github/workflows/spellcheck.yml]: No it wasn't. Copy paste failure from the [example on the repo](https://github.com/insightsengineering/r-spellcheck-action?tab=readme-ov-file#usage)

corrected with 4933aa5

Comment (averissimo) [reply .github/workflows/spellcheck.yml]: However, it seems like `v3` has a bug. I've submitted a bug there: https://github.com/insightsengineering/r-spellcheck-action/issues/9

Comment (averissimo) [reply .github/workflows/spellcheck.yml]: Solved upstream

File: DESCRIPTION
Comment (bms63) [inline DESCRIPTION:1]: wondering if we have the ADSL already stored as a data object in xportr if we need admiral in suggests??

Comment (averissimo) [reply DESCRIPTION:1]: Sounds good, the vignettes that have `library(admiral)` all build without `{admiral}`

It's a quick implementation.

File: .github/workflows/check-standard.yaml [hunk 5]
```diff
@@ -25,24 +38,29 @@ jobs:
           - { os: ubuntu-latest, r: "devel", http-user-agent: "release" }
```
Comment (bms63) [inline .github/workflows/check-standard.yaml:38]: shall we add a windows test in the matrix?

File: .github/workflows/common.yaml [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+  # to run this workflow on demand, anytime
+  workflow_dispatch:
+
+concurrency:
+  group: common-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  coverage:
+    name: Code Coverage (admiral)
```
Comment (bms63) [inline .github/workflows/common.yaml:32]: should we remove the admiral reference?

**Review by bms63 — COMMENTED** [review]
Hi @averissimo just checking on decisions with this PR - do you need any other help or resolutions?

File: .github/workflows/common.yaml
[the remark this answers is not in the document]
Comment (averissimo) [reply .github/workflows/common.yaml:1]: Common file seems like the way to go.

I created the `badges` branch just to allow for (basically) the same workflow file to be used in {xportr}

**Comment by averissimo** [comment]
Kept our CMD workflow as it allows for multiple platforms.

And it's a better test than insightsengineer custom docker environment

**Comment by bms63** [comment]
LGTM!  many thanks again!!

**Pull request merged**